File: paper_artifacts/experiments/llr9/artifacts/sources/kimi27sglang-c/fuse_move_ifs__submitted.c
#include <stddef.h>
#include <stdint.h>

void fuse_move_ifs_fp64(double * restrict a, double * restrict b, const double * restrict cond, const double * restrict src, int64_t K, int64_t LEN_2D, uint8_t * restrict workspace, int64_t workspace_bytes) {
    if (K > 0) {
        #pragma omp parallel for
        for (int64_t i = 0; i < LEN_2D; i++) {
            if (cond[i] > 0.0) {
                for (int64_t j = 0; j < LEN_2D; j++) {
                    double v = src[i * LEN_2D + j];
                    a[i * LEN_2D + j] = v * 2.0;
                    b[i * LEN_2D + j] = v + 1.0;
                }
            } else {
                for (int64_t j = 0; j < LEN_2D; j++) {
                    b[i * LEN_2D + j] = src[i * LEN_2D + j] + 1.0;
                }
            }
        }
    } else {
        #pragma omp parallel for
        for (int64_t i = 0; i < LEN_2D; i++) {
            if (cond[i] > 0.0) {
                for (int64_t j = 0; j < LEN_2D; j++) {
                    a[i * LEN_2D + j] = src[i * LEN_2D + j] * 2.0;
                }
            }
        }
    }
}

File: paper_artifacts/experiments/llr8/artifacts/sources/qwen38-fortran-skills/fuse_move_ifs__submitted.f90
subroutine fuse_move_ifs_fp64(a, b, cond, src, k, len_2d, workspace, workspace_size) bind(C)
  use, intrinsic :: iso_c_binding
  implicit none
  integer(c_int64_t), value, intent(in) :: k, len_2d, workspace_size
  real(c_double), intent(inout) :: a(len_2d, len_2d)
  real(c_double), intent(inout) :: b(len_2d, len_2d)
  real(c_double), intent(in)    :: cond(len_2d)
  real(c_double), intent(in)    :: src(len_2d, len_2d)
  type(c_ptr), value :: workspace

  integer(c_int64_t) :: i, j

  if (k > 0) then
    ! Fused single pass over src: b is unconditional, a only where cond>0.
    ! src read exactly once. Row i independent -> thread the outer loop.
    !$omp parallel do
    do i = 1, len_2d
      if (cond(i) > 0.0d0) then
        do j = 1, len_2d
          b(j, i) = src(j, i) + 1.0d0
          a(j, i) = src(j, i) * 2.0d0
        end do
      else
        do j = 1, len_2d
          b(j, i) = src(j, i) + 1.0d0
        end do
      end if
    end do
  else
    !$omp parallel do
    do i = 1, len_2d
      if (cond(i) > 0.0d0) then
        do j = 1, len_2d
          a(j, i) = src(j, i) * 2.0d0
        end do
      end if
    end do
  end if
end subroutine fuse_move_ifs_fp64
